quickwit-search/src/lib.rs

```diff
@@ -67,10 +67,13 @@ pub use crate::search_stream::root_search_stream;
 pub use crate::service::{MockSearchService, SearchService, SearchServiceImpl};
 use crate::thread_pool::run_cpu_intensive;
 
-/// Compute the gRPC port from the SWIM port.
-/// Add 1 to the SWIM port to get the gRPC port.
-pub fn swim_addr_to_grpc_addr(swim_addr: SocketAddr) -> SocketAddr {
-    SocketAddr::new(swim_addr.ip(), swim_addr.port() + 1)
+/// Compute the gRPC port from the Scuttlebutt gossip port.
+/// Add 1 to the Scuttlebutt gossip port to get the gRPC port.
+pub fn scuttlebutt_gossip_addr_to_grpc_addr(scuttlebutt_gossip_addr: SocketAddr) -> SocketAddr {
+    SocketAddr::new(
```

Comment:
Can you open a ticket to fix our port story?

Right now we hardcoded udp -> grpc because we did not have any way to advertise it.
Now we can use scuttlebut to let other nodes know what is our grpc port.

Spec should be:
- rest port /  gossip port / grpc port are configured in the conf
- rest port has a hardcoded default value that can be overriden
- grpc and gossip port default value are computed from the rest port
- clients use scuttlebutt to know the grpc port of their peers.